src/actions.py

```diff
@@ -49,7 +49,7 @@ def on_rotate_credentials(self, event: ops.ActionEvent) -> None:
             event: The rotate credentials event.
         """
         container = self.charm.unit.get_container(JENKINS_SERVICE_NAME)
-        if not container.can_connect() or not jenkins.is_storage_ready(container):
+        if not jenkins.is_storage_ready(container):
```

Comment:
I wonder if this couldn't be an Exception raised by rotate_credentials instead of checking this here.

Reply:
Perhaps, but I'd rather keep the cahnges as contained as possible for this PR. This seems a bit out of the scope of the PR

Reply:
No worries, just caught my attention because the container.can_connect was moved to is_storage_ready.